src/extensionFunctions.ts

```diff
@@ -896,4 +905,35 @@ export function sendTextToNamedOutputChannel(message: string, channelName?: stri
 	if (!channelName || channelName === DIDACT_OUTPUT_CHANNEL) {
 		didactOutputChannel.show();
 	}
-}
\ No newline at end of file
+}
+
+export async function copyFileTextToClipboard(uri: vscode.Uri) : Promise<void> {
+	const testUri : vscode.Uri = uri;
+	const out : vscode.Uri | undefined = handleVSCodeDidactUriParsingForPath(testUri);
+	if (!out) {
+		const errmsg = `ERROR: No file found when parsing path ${uri}`;
+		sendTextToOutputChannel(errmsg);
+		vscode.window.showErrorMessage(errmsg);
+		throw new Error(errmsg);
+	} else {
+		let content : string | undefined = undefined;
+		if (out.scheme === 'file') {
+			try {
+				content = fs.readFileSync(out.fsPath, 'utf8');
```

Comment:
There is also the VSCode internal implementation of the fs library which is not blocking. Maybe you can take a look if it is possible to use it instead of the external one?

Reply:
I have looked at that in the past and so far I have only found the internal library useful when simply checking to see if we can get file contents, but not in actually retrieving the file contents. For instance https://github.com/redhat-developer/vscode-didact/blob/e03cf0bd48217b5c24134de7ec2e354cf3d72be7/src/extensionFunctions.ts#L735 just uses the internal implementation to make sure we can get something out of the file, but we're using the fs.readFileSync call most places to get the actual data out as a UTF-8 encoded string. 

I have not found another useful way to play with that internal fs version yet.

Reply:
@lhein provided an example

```
const f = await vscode.workspace.fs.readFile(vscode.Uri.file(file));
const yamlDoc:DataVirtConfig = YAML.parse(f.toString());
```

I will spin this off into a separate JIRA to explore, since we use the external FS in many places and it might be useful to do a migration.